common/sign/v1/iss.c.inc

```diff
@@ -106,13 +106,14 @@ int _ISS_PREFIX(sig_digest)(trit_t *const dig, trit_t const *const hash,
                             HASH_STATE *const state) {
   size_t i, j;
   trit_t *sig_start = sig, *sig_end = &sig[sig_len];
-  for (i = 0; sig < sig_end; i++, sig = &sig[HASH_LENGTH_TRIT]) {
-    for (j = 0; j < (HASH_TRYTE_VAL(hash, i) - TRYTE_VALUE_MIN); j++) {
+  for (i = 0, sig = sig_start; sig < sig_end; i++, sig = &sig[HASH_LENGTH_TRIT]) {
+    for (j = (hash[i] - TRYTE_VALUE_MIN); j-- > 0;) {
```

Comment:
No, `hash` is a trit array and not a tryte array, hence the need for the `HASH_TRYTE_VAL`m acro.

Reply:
this is main change I did. I looked up java IRI ISSinPlace.digest() implementation
java code is doing in this way.

Reply:
IRI is doing it this way just because it takes a trytes hash as parameter.
Entangled takes a trits hash, that's why we have we have the macro.
Still, both do exactly the same thing in a different way.

Reply:
I think it is using trits instead. java test is here,
https://github.com/iotaledger/iri/pull/1368

Reply:
It's not. It doesn't make sense to do `hash[i] - TRYTE_VALUE_MIN` if hash[i] is a trit. It would mean doing {-1, 0, 1} - (-13),  which is not what we want.